plugins/transfer/transfer.plugin.zsh

```diff
@@ -0,0 +1,20 @@
+function transfer() {
+  if [ $# -eq 0 ]; then
+    echo "No arguments specified. Usage:\necho transfer /tmp/test.md\ncat /tmp/test.md | transfer test.md"
+    return 1
+  fi
+
+  tmpfile=$( mktemp -t transferXXX )
+  if tty -s; then
+    basefile=$(basename "$1" | sed -e 's/[^a-zA-Z0-9._-]/-/g')
```

Comment:
You can replace `basename` and `sed` with native zsh parameter expansion (e.g. `${foo:t}` for basename). But since you're using this in a URL, you might want to replace the simple `sed` call with actual URL-escaping using the new `omz_urlencode()` function from #4129. That could get you nicer URLs and avoid collisions for files with special characters in their names.